### ISSUE TITLE
HTTP Request Tool

### ISSUE THREAD

**Issue by iGamerTrip** [CorentinTh/it-tools#837]
**What tool do you want?**
Example: an HTTP request tool

**Describe the solution you'd like**
A simple, powerful, and easily shareable HTTP request tool, where you can also export the request in curl format, or even a few popular languages

**Is their example of this tool in the wild?**
[Cyberchef](https://gchq.github.io/CyberChef/#recipe=HTTP_request('GET','','','Cross-Origin%20Resource%20Sharing',false)) has a good implementation, but I found it doesn't work very well with some requests for some reason, namely Cloudflare API calls

**Additional context**
Having a "share" button maybe nice, along with a warning if you had any authorization headers to avoid leaking your tokens, but that's not a requirement for it to be an amazing tool

**Comment by sharevb** [comment]
Hi @iGamerTrip, this is not really possible from a browser, JS cannot access everywhere, will require a local application or server side, so probably not the goal of this project

**Comment by iGamerTrip** [comment]
While i dont know a lot about JS, I've noticed that Cyberchef does have that functionality from the browser, and so does Postman

It also doesn't necessarily need to be able to access everywhere, just public APIs and the like
Postman seems to suggest `fetch()` for JS, and here's an example code snippet it provided:
```
const requestOptions = {
  method: "GET",
  redirect: "follow"
};

fetch("https://api.sampleapis.com/switch/games", requestOptions)
  .then((response) => response.text())
  .then((result) => console.log(result)) // Returns a list of Switch games!
  .catch((error) => console.error(error));
  ```
  
  I apologize if this is simply unfeasible, or if I'm misunderstanding things, as my area of expertise lies moreso in Python than JS, but I thought this might help get my point across

**Comment by sharevb** [comment]
Hi, yes, for api calls, (supposing CORS policy is ok), but will not work for websites as JS is not allowed to call everywhere (security)

By the way curl generator and translation to langages is possible

**Comment by iGamerTrip** [comment]
Yeah, a better name for this issue is probably "API Request Tool", as it'd be good to test APIs (with proper CORS policies) with
And a curl/popular languages generator would be really cool as well!

**Comment by sharevb** [comment]
HI @Trip7274 and @CorentinTh, implemented i #1168

**Comment by Trip7274** [comment]
Thanks a ton!

**Comment by sharevb** [comment]
Hi @Trip7274 , implemented: https://sharevb-it-tools.vercel.app/api-tester

And if you are interested in an up to date version of it-tools, with many improvements, new tools, and bug fixes, as this repo is almost no more maintained, I made a fork here : https://github.com/sharevb/it-tools (https://sharevb-it-tools.vercel.app/ and docker images https://github.com/sharevb/it-tools/pkgs/container/it-tools)
